### PR TITLE
ft(intergration) Create roles and set permissions

### PR DESCRIPTION
**Pull request by wakaflorien**
### What this PR does 

> This pr gives admin the ability to create roles and set permissions to those roles.

- [x] The **Administrator** should be able to create, update, and delete roles.
- [x] The **administrator** should be able to set and update role's permissions.
- [x] The Access control has been implemented.
- [x] Appropriate error messages are displayed on unsuccessful attempts

## Trello card Id

#TP-15

## How you can manually test it

**Clone the repo**
then checkout to **ft-create-roles-and-permissions-integration-TP-15** branch

**Install all dependencies using **
```
npm i

```

**run development server using**

```
npm run dev
```

**Development links for docker**
[Frontend](http://localhost:8080/) 
[Backend](http://localhost:5000/)

**Development links for dev server (npm run dev)**
[Frontend](http://localhost:8080/)

## Trello
https://trello.com/c/EwSbsIZ3

### Screenshots
Screenshot from 2022-05-04 10-15-32
Screenshot from 2022-05-04 10-15-35
Screenshot from 2022-05-04 10-15-51
Screenshot from 2022-05-04 10-15-56
Screenshot from 2022-05-04 10-16-03
Screenshot from 2022-05-04 10-16-07

Screenshot from 2022-05-04 10-15-22

### CODE REVIEW

**Review by sezeranoJchrisostome — CHANGES_REQUESTED** [review]
Well done @wakaflorien, now I see the state on permissions assign are reflected from the backend 
Can we reduce the pop-ups that are revealed on assign?

File: src/__test__/reducers/rolesReducer.test.js [hunk 1]
```diff
@@ -100,7 +87,7 @@ it("should accept the deletion of permission", () => {
     const state = [
         {
             id: 1,
-            name: "Admin",
+            rolename: "Admin",
```
Comment (renepromesse) [inline src/__test__/reducers/rolesReducer.test.js:90]: why are you changin the property from role to rolename?

**Review by renepromesse — CHANGES_REQUESTED** [review]
hey @wakaflorien , nice work

- could you add the screenshots?

File: src/__test__/reducers/rolesReducer.test.js (hunk 1 above)
Comment (wakaflorien) [reply src/__test__/reducers/rolesReducer.test.js:90]: because that's the name of a field in db

**Review by chanceDira — DISMISSED** [review]
Great work @wakaflorien

**Review by jeandamascyifuzo — COMMENTED** [review]
well done @wakaflorien 
could you please add a screen-shot to compare my result to yours

**Comment by wakaflorien** [comment]
> 

yes will do.

**Review by sezeranoJchrisostome — APPROVED** [review]

**Review by renepromesse — APPROVED** [review]
Great @wakaflorien

**Review by jeandamascyifuzo — APPROVED** [review]

**Pull request merged**